### DISCUSSION TITLE
Additional Laravel 11 generators

### DISCUSSION THREAD

**Discussion by jaspertey** [lunarstorm/laravel-ddd#53] [Ideas]
Noting this down as a plan to add support for the rest of the new commands that were introduced in Laravel 11: https://laravel.com/docs/11.x/releases#new-artisan-commands

```bash
php artisan ddd:class
php artisan ddd:interface
php artisan ddd:trait
```

v1.0.0 added `ddd:enum`, but not the three above.

**Comment by jaspertey** [comment]
Implemented in #55

**Reply by looqey** [reply]
Hey! I wanted to ask you a question: why these commands couldn't be added to Laravel 10.25+ package version?

**Reply by jaspertey** [reply]
> Hey! I wanted to ask you a question: why these commands couldn't be added to Laravel 10.25+ package version?

Internally, these particular ddd commands extend Laravel's equivalent native commands, and they are only available in Laravel 11. To make it work in this package for Laravel 10, separate generators would have to be written only for Laravel 10 - which seemed rather counterproductive vs. users making an effort to upgrade to Laravel 11. That was the decision at the time, anyway.
